tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -0,0 +1,225 @@
+
+    Returns:
+      A response that contains a JSON object. The keys of the object
+      are all the runs. Each run is mapped to a (potentially empty) dictionary
+      whose keys are tags associated with run and whose values are metadata
+      (dictionaries).
+
+      The metadata dictionaries contain 2 keys:
+        - displayName: For the display name used atop visualizations in
+            TensorBoard.
+        - description: The description that appears near visualizations upon the
+            user hovering over a certain icon.
+    """
+    return http_util.Respond(
+        request, self.tags_impl(), 'application/json')
+
+  def tags_impl(self):
+    """Creates the JSON object for the tags route response.
+
+    Returns:
+      The JSON object for the tags route response.
+    """
+    runs = self._multiplexer.Runs()
+    result = {run: {} for run in runs}
+
+    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
+    for (run, tag_to_content) in six.iteritems(mapping):
+      for (tag, content) in six.iteritems(tag_to_content):
+        summary_metadata = self._multiplexer.SummaryMetadata(run, tag)
+        result[run][tag] = {'displayName': summary_metadata.display_name,
+                            'description': plugin_util.markdown_to_safe_html(
+                                summary_metadata.summary_description)}
+
+    return result
+
+  @wrappers.Request.application
+  def available_time_entries_route(self, request):
+    """Gets a dict mapping run to a list of time entries.
+    Returns:
+      A dict with string keys (all runs with PR curve data). The values of the
+      dict are lists of time entries (consisting of the 3 fields below) to be
+      used in populating values within time sliders.
+    """
+    return http_util.Respond(
+        request, self.available_time_entries_impl(), 'application/json')
+
+  def available_time_entries_impl(self):
+    """Creates the JSON object for the available time entries route response.
+    Returns:
+      The JSON object for the available time entries route response.
+    """
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        PrCurvesPlugin.plugin_name)
+    response = {}
+
+    # Compute the max step exhibited by any tag for each run. If a run lacks
+    # data, exclude it from the mapping.
+    for run, tag_to_content in all_runs.items():
+      if not tag_to_content:
+        # This run lacks data for this plugin.
+        continue
+
+      # Just use the list of tensor events for any of the tags to determine the
+      # steps to list for the run. The steps are often the same across tags for
+      # each run, albeit the user may elect to sample certain tags differently
+      # within the same run. If the latter occurs, TensorBoard will show the
+      # actual step of each tag atop the card for the tag.
+      tensor_events = self._multiplexer.Tensors(
+          run, list(tag_to_content.keys())[0])
```

Comment:
Thank you for noticing! I undid the rogue changes in a recent commit. What happened was that I had copied some code from the PR with the demo, which did not include some of the recent changes in this particular PR. I need to be more cognizant of avoiding rogue changes and commits.